lib/charms/ip_router_interface/v0/ip_router_interface.py

```diff
@@ -29,4 +132,273 @@
 # to 0 if you are raising the major API version
 LIBPATCH = 1
 
-# TODO: add your code here! Happy coding!
+from ipaddress import IPv4Address, IPv4Network
+from copy import deepcopy
+from typing import Dict, List, Union, TypeAlias
+from ops.framework import Object, EventSource, EventBase, ObjectEvents
+from ops.charm import CharmBase
+from ops import RelationJoinedEvent, RelationChangedEvent, RelationDepartedEvent, StoredState
+import logging, json
+
+logger = logging.getLogger(__name__)
```

Comment:
I find this lib to be lacking logging. As far as I see, we log nothing

Reply:
Logged a bunch of stuff